satellite/console/consoleweb/consoleql/mutation.go

```diff
@@ -420,6 +424,95 @@ func rootMutation(log *zap.Logger, service *console.Service, mailService *mailse
 					return keys, nil
 				},
 			},
+			AddPaymentMethodMutation: &graphql.Field{
+				Type: graphql.Boolean,
+				Args: graphql.FieldConfigArgument{
+					FieldProjectID: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldCardToken: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldIsDefault: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.Boolean)},
+				},
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					projectID, _ := p.Args[FieldProjectID].(string)
+					cardToken, _ := p.Args[FieldCardToken].(string)
+					isDefault, _ := p.Args[FieldIsDefault].(bool)
+
+					auth, err := console.GetAuth(p.Context)
+					if err != nil {
+						return false, err
+					}
+
+					projID, err := uuid.Parse(projectID)
+					if err != nil {
+						return false, err
+					}
+
+					_, err = service.AddNewPaymentMethod(p.Context, cardToken, isDefault, *projID, auth.User.ID)
+					if err != nil {
+						return false, err
+					}
+
+					return true, nil
+				},
+			},
+			DeletePaymentMethodMutation: &graphql.Field{
+				Type: graphql.Boolean,
+				Args: graphql.FieldConfigArgument{
+					FieldID: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+				},
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					fieldProjectPaymentID, _ := p.Args[FieldID].(string)
+
+					_, err := console.GetAuth(p.Context)
+					if err != nil {
+						return false, err
+					}
+
+					paymentID, err := uuid.Parse(fieldProjectPaymentID)
+					if err != nil {
+						return false, err
+					}
+
+					err = service.DeletePaymentMethod(p.Context, *paymentID)
+					if err != nil {
+						return false, err
+					}
+
+					return true, nil
+				},
+			},
+			SetDefaultPaymentMethodMutation: &graphql.Field{
+				Type: graphql.Boolean,
+				Args: graphql.FieldConfigArgument{
+					FieldProjectID: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldID:        &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+				},
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					fieldProjectID, _ := p.Args[FieldProjectID].(string)
+					fieldProjectPaymentID, _ := p.Args[FieldID].(string)
+
+					_, err := console.GetAuth(p.Context)
```

Comment:
Why checking auth here?

Reply:
To prevent it from going to deeper layer to console/service